nomad/structs/structs.go

```diff
@@ -8411,6 +8434,27 @@ func (p *Plan) IsNoOp() bool {
 		len(p.DeploymentUpdates) == 0
 }
 
+func (p *Plan) NormalizeAllocations() {
```

Comment:
comment

Reply:
@schmichael I'll take a look at this probably in the next week. But, I think in the current shape of things, this isn't possible. I think the flaw is that currently we don't have a separate concept of DTOs, when we send the object over the wire, which is something I did point out in the design discussions. It's not that I don't want to use more strongly typed code, I do and I did explore it. But, switching to DTOs was more work, and we didn't scope it.